front/app/components/EsriMap/index.tsx

```diff
@@ -208,19 +236,60 @@ const EsriMap = ({
 
       initialValuesLoaded.current = true;
     }
-  }, [globalMapSettings, initialData, isMobileOrSmaller, map, mapView]);
+  }, [
+    globalMapSettings,
+    initialData,
+    isMobileOrSmaller,
+    map,
+    mapView,
+    webMapId,
+  ]);
 
   // Load dynamic data that was passed in.
   // Note: This data is dynamic and may change.
   useEffect(() => {
     // Add any map layers which were passed in
+
+    // Handle when we're using a webmap
+    webMap?.when(() => {
+      if (webMap && layers) {
+        // If there are any Web Map reference layers, re-order so they are below other layers created in our application
+        if (referenceLayers && referenceLayers.length > 0) {
+          const newBasemapLayers = webMap.basemap.baseLayers;
+          webMap.addMany(referenceLayers.toArray());
+          webMap.basemap = new Basemap({
+            baseLayers: newBasemapLayers,
+          });
+        }
+
+        // Add any layers passed in as props
+        layers.forEach((layer) => {
+          webMap.add(layer);
+        });
+
+        // If we have WebMap reference layers, save them in state
+        const refLayers =
+          webMap.basemap.referenceLayers.length > 0
+            ? webMap.basemap.referenceLayers
+            : undefined;
+
+        if (refLayers?.length && refLayers.length > 0) {
+          setReferenceLayers(refLayers);
+        }
+      }
+    });
+
+    // Handle layers for the default map (when we're not using a Web Map)
```

Comment:
@luucvanderzee Actually, it always runs whenever a layer is added. Since a **_WebMap_** and the default **_Map_** are two separate objects, if we add a WebMap + add more Layers to that map, and then choose to remove the WebMap, we still want all the added Feature Layers to show on the normal Map once the WebMap is removed.

So essentially, whenever a layer is added to a WebMap, we also add it to the Map at the same time, so it's there if we need it :) The layers aren't actually fetched from ArcGIS Online unless the map is visible on the screen, so this doesn't add any overhead or anything.

Reply:
I'm going to update the comment as well so that's clear - my wording is indeed confusing.